developer/src/kmc-keyboard/src/compiler/disp.ts

```diff
@@ -0,0 +1,62 @@
+import { constants } from "@keymanapp/ldml-keyboard-constants";
+import { KMXPlus } from '@keymanapp/common-types';
+
+import { CompilerMessages } from "./messages.js";
+import { SectionCompiler } from "./section-compiler.js";
+
+import GlobalSections = KMXPlus.GlobalSections;
+import Disp = KMXPlus.Disp;
+import DispItem = KMXPlus.DispItem;
+
+export class DispCompiler extends SectionCompiler {
+
+  public get id() {
+    return constants.section.disp;
+  }
+
+  public validate(): boolean {
+    let valid = true;
+
+    const tos = new Set();
+
+    if (this.keyboard.displays?.display) {
+        for (const { to } of this.keyboard.displays?.display) {
+            if (tos.has(to)) {
+                this.callbacks.reportMessage(CompilerMessages.Error_DisplayIsRepeated({ to }));
+                return false;
+            }
+            tos.add(to);
+        }
+    }
+
+    return valid;
+  }
+
+  public compile(sections: GlobalSections): Disp {
+    let result = new Disp();
+
+    // displayOptions
+    result.baseCharacter = sections.strs.allocString(this.keyboard.displays?.displayOptions?.baseCharacter)
+
+    // displays
+    result.disps = this.keyboard.displays?.display.map(display => ({
+        to: sections.strs.allocString(display.to),
+        display: sections.strs.allocString(display.display),
+    }));
```

Comment:
I think `result.disps` should be `[]` if `this.keyboard.displays` is nullish. If that can't happen we shouldn't be using the `?.` operator :grin:

Reply:
done this now